test/e2e/apps/statefulset.go

```diff
@@ -588,6 +589,29 @@ var _ = SIGDescribe("StatefulSet", func() {
 			})
 			framework.ExpectNoError(err)
 
+			// Verify that statuful set will be scaled up in order.
+			wg := sync.WaitGroup{}
+			var orderErr error
+			wg.Add(1)
+			go func() {
+				defer wg.Done()
+
+				expectedOrder := []string{ssName + "-0", ssName + "-1", ssName + "-2"}
+				ctx, cancel := watchtools.ContextWithOptionalTimeout(context.Background(), statefulSetTimeout)
```

Comment:
the timeout here probably isn't right, now that this starts before the statefulset does stuff

Reply:
It's not the matter of timeout - the 10 minut timeout seems fine to me.

It was a matter of waiting for Added" event in the second case.